src/Resources/views/standard_layout.html.twig

```diff
@@ -66,22 +66,36 @@ file that was distributed with this source code.
             {% endblock %}
 
             {% set locale = app.request.locale %}
-            {# localize moment #}
+
+            {# omit default "en" locale #}
             {% if locale[:2] != 'en' %}
+                {# localize moment #}
+                {% set localeForMoment = locale|lower|replace({'_':'-'}) %}
+                {# `moment: ^2.8` only ships "es" and "es-do" locales for "es" language #}
+                {% if localeForMoment[:2] == 'es' and localeForMoment not in ['es', 'es-do'] %}{% set localeForMoment = 'es' %}{% endif %}
+                {# @todo: there are more locales which are not supported by moment and they need to be translated/normalized/canonicalized here #}
                 <script src="{{ asset(
                     'bundles/sonatacore/vendor/moment/locale/' ~
-                    locale|lower|replace({'_':'-'}) ~
+                    localeForMoment ~
                     '.js'
                 ) }}"></script>
-            {% endif %}
-
-            {# localize select2 #}
-            {% if sonata_admin.adminPool.getOption('use_select2') %}
-                {% if locale == 'pt' %}{% set locale = 'pt_PT' %}{% endif %}
 
-                {# omit default EN locale #}
-                {% if locale[:2] != 'en' %}
-                    <script src="{{ asset('bundles/sonatacore/vendor/select2/select2_locale_' ~ locale|replace({'_':'-'}) ~ '.js') }}"></script>
+                {# localize select2 #}
+                {% if sonata_admin.adminPool.getOption('use_select2') %}
```

Comment:
Can we move this stuff to a twig function to reduce the code inside the template?

Reply:
with tests, preferrably